ebos/equil/initstateequil.hh

```diff
@@ -1928,6 +1948,26 @@ private:
             saturations = psat.deriveSaturations(pos, eqreg, ptable);
             pressures   = psat.correctedPhasePressures();
 
+            // TODO: not totally sure this is the cartesian Index
+            const auto global_index = cartesianIndexMapper_.cartesianIndex(cell);
+            if (!num_aqu_cells.empty()) {
+                const auto search = num_aqu_cells.find(global_index);
+                if (search != num_aqu_cells.end()) {
+                    saturations = {0.0, 0.0, 1.0};
+                    const auto* aqu_cell = num_aqu_cells.at(global_index);
+                    const auto msg = fmt::format("FOR AQUIFER CELL AT ({}, {}, {}) OF NUMERICAL "
+                                                 "AQUIFER {}, WATER SATURATION IS SET TO BE UNITY",
+                                                 aqu_cell->I+1, aqu_cell->J+1, aqu_cell->K+1, aqu_cell->aquifer_id);
+                    OpmLog::info(msg);
+
+                    if (aqu_cell->init_pressure) {
+                        const double pres = *(aqu_cell->init_pressure);
+                        pressures = {pres, pres, pres};
+                    }
+                }
+
+            }
```

Comment:
I suppose this bit could be split out to a separate helper function that we'd run at the end of the initialisation step.  That way it'd be applicable to all equilibration methods (i.e., cell-centre based, horizontal and, when we get it, tilted).

Reply:
I addressed it in the new commit. Please check whether it is something you want.